server/src/main/java/org/spine3/server/command/CommandScheduler.java

```diff
@@ -20,32 +20,83 @@
 
 package org.spine3.server.command;
 
+import com.google.protobuf.Duration;
+import com.google.protobuf.Timestamp;
+import io.grpc.stub.StreamObserver;
 import org.spine3.base.Command;
+import org.spine3.base.CommandContext;
 import org.spine3.base.CommandId;
+import org.spine3.base.Response;
+import org.spine3.base.Responses;
+import org.spine3.envelope.CommandEnvelope;
 
 import javax.annotation.CheckReturnValue;
 import javax.annotation.Nullable;
 import java.util.Set;
 
+import static com.google.common.base.Preconditions.checkNotNull;
 import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.collect.Sets.newHashSet;
+import static com.google.protobuf.util.Timestamps.checkValid;
+import static org.spine3.base.CommandStatus.SCHEDULED;
 import static org.spine3.base.Commands.getId;
-import static org.spine3.base.Commands.setSchedulingTime;
+import static org.spine3.base.Commands.isScheduled;
 import static org.spine3.protobuf.Timestamps2.getCurrentTime;
 
 /**
  * Schedules commands delivering them to the target according to the scheduling options.
  *
  * @author Alexander Litus
+ * @author Alexander Yevsyukov
  */
-public abstract class CommandScheduler {
+public abstract class CommandScheduler implements CommandBusFilter {
 
     private static final Set<CommandId> scheduledCommandIds = newHashSet();
 
     private boolean isActive = true;
 
+    @Nullable
     private CommandBus commandBus;
 
+    @Nullable
+    private Rescheduler rescheduler;
+
+    protected CommandScheduler() {
+    }
+
+    public void setCommandBus(CommandBus commandBus) {
```

Comment:
It's going to be `protected`. We need this method to inject the Command Bus into scheduler when the bus is built.